engine/servermaster/executor_manager.go

```diff
@@ -285,9 +285,11 @@ func (e *ExecutorManagerImpl) ExecutorCount(status model.ExecutorStatus) (count
 	e.mu.Lock()
 	defer e.mu.Unlock()
 	for _, executor := range e.executors {
+		executor.mu.RLock()
```

Comment:
It seems that `Executor` is in the boundary of `ExecutorManagerImpl`, which means the managerment and lock protection should be the responsible of `ExecutorManagerImpl`.

Any case show that the state of `Executor` will be updated independently?

Reply:
https://github.com/pingcap/tiflow/blob/d8afd07f45f5f38be1d89ccb3447b4aa70b44177/engine/servermaster/executor_manager.go#L131-L140

Reply:
Can you wrap these logic to the method of `Executor`?
It's easy for developer not hold the `exec.mu` when update the status of `Executor`.